project/backend/src/test/java/com/google/sps/TestRestaurantParser.java

```diff
@@ -0,0 +1,43 @@
+package com.google.sps;
+
+import com.google.sps.data.JsonToRestaurantParser;
+import com.google.sps.data.Restaurant;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public final class TestRestaurantParser {
+
+  private static final int MISSING_FIELD = -1;
+  private static final List<String> EMPTY_TYPES = new ArrayList<>();
+  private static final String RESTAURANT_A = "RESTAURANT_A";
+  private static final String PLACE_ID_A = "A12345";
+  private static final String VICINITY_A = "10 Main Street";
+  private static final Map<String, Double> COORDS_A = Map.ofEntries(
+    Map.entry("lat", 10.0),
+    Map.entry("lng", 20.0)
+  );
+
+  @Test
+  public void noOptionalFields() throws JSONException {
+    JSONObject body = new JSONObject(
+      "{'geometry' : {'location' : {'lat' : " + COORDS_A.get("lat") + ", 'lng' : " + COORDS_A.get("lng") + "}}," + 
+      "'place_id' : " + PLACE_ID_A + 
+      ",'name' : " + RESTAURANT_A + 
+      ",'vicinity' : " + VICINITY_A + 
+      ",'types' : [],}"
+    );
```

Comment:
Not sure if I'm happy with the way I initialized this... thinking using the way they did it in 4.1 here may be better https://www.baeldung.com/java-org-json#1-creating-json-directly-from-jsonobject